distributed_nose/plugin.py

```diff
@@ -88,6 +127,73 @@ def configure(self, options, config):
             # let's distribute their tests
             self.enabled = True
 
+        if self.algorithm == self.ALGORITHM_LEAST_PROCESSING_TIME:
+            if not self.lpt_data_filepath:
+                logger.critical((
+                    "'--algorithm least-processing-time' requires "
+                    "'--lpt-data <lpt-data-filepath>' to be specified as "
+                    "well. Falling back to hash-ring algorithm."
+                ))
+                self.algorithm = self.ALGORITHM_HASH_RING
```

Comment:
thoughts on just crashing here+below instead of falling back? my guess is it's intended to defend against things higher up in the stack (eg failing to download a data file), which does seem like what we want but might not be the best place to handle it

Reply:
right, good point. lemme give that some thought.

Reply:
actually, yeah, falling back is definitely bad in the case of the failed download since that might have happened at only one node and then the different nodes might successfully execute different algorithms.

good call. i'll revise this to just crash and fail for these cases.

Reply:
ah, yeah, good point. fwiw, so long as nose doesn't write out a nosetests.xml our builds will consider it unstable, so we could hook up retries at that level if needed.